pkg/resource/sensitive.go

```diff
@@ -199,3 +219,37 @@ func expandedFor(expanded fieldpath.Segments, withWildcard fieldpath.Segments) b
 func normalizeJSONPath(s string) string {
 	return strings.TrimPrefix(strings.TrimPrefix(s, "spec.forProvider."), "status.atProvider.")
 }
+
+func secretKeyToFieldPath(s string) (string, error) {
+	s1 := reInsideThreeDotsBlock.ReplaceAllString(s, "[$1]")
+	s2 := reEndsWithIndex.ReplaceAllString(s1, "[$1]")
+	s3 := reMiddleIndex.ReplaceAllString(s2, "[$1].")
+	seg, err := fieldpath.Parse(s3)
+	if err != nil {
+		return "", errors.Wrapf(err, "cannot parse secret key %q as fieldpath", s3)
+	}
+	return seg.String(), nil
+}
+
+func fieldPathToSecretKey(s string) (string, error) {
+	sg, err := fieldpath.Parse(s)
+	if err != nil {
+		return "", errors.Wrapf(err, "cannot parse %q as fieldpath", s)
+	}
+
+	var b strings.Builder
+	for _, s := range sg {
+		switch s.Type {
+		case fieldpath.SegmentField:
+			if strings.ContainsRune(s.Field, '.') {
+				b.WriteString(fmt.Sprintf("...%s...", s.Field))
```

Comment:
I don't love having three dots here but seems like we don't have another choice.